### PR TITLE
fix(pgwire) Rudimentary support for COPY FROM via pgwire

### PR DESCRIPTION
**Pull request by jerrinot**
The code was mostly there, but [this] was throwing NPE due to the missing text loader. 

It does not support for reading from stdin,
but it works with a CSV file local to qdb

### CODE REVIEW

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java [hunk 1]
```diff
@@ -1424,6 +1424,11 @@ private void parse(long address, int len, @Transient SqlCompiler compiler)
             case 'Q':
                 processQuery(msgLo, msgLimit, compiler);
                 break;
```
Comment (bziobrowski) [inline core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:1426]: Is this CopyDone message handler required for jdbc or other clients ?

Comment (jerrinot) [reply core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:1426]: this line sends it: https://github.com/questdb/questdb/pull/2061/files#diff-8a715db0fe34387362d99b2bf088facf1faa2c5d0c6192abd0a2e7cd8d1e1fc1R1693

I could not figure out how to use COPY without the `CopyManager` API. Just using `statement.query(copy ...)` does not work for me. I can try whether it works with Postgres.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 3 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/CompiledQueryImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|1|1|100.00%|

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java [hunk 2]
```diff
@@ -1844,7 +1844,6 @@ private void processCompiledQuery(SqlCompiler compiler, CompiledQuery cq)
             case CompiledQuery.COPY_LOCAL:
                 // uncached
                 queryTag = TAG_COPY;
-                sendCopyInResponse(compiler.getEngine(), cq.getTextLoader());
```
Comment (jerrinot) [inline core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:1847]: sending this confuses clients. they do not expect the copy subprotocol response.

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java (hunk 1 above)
Comment (jerrinot) [reply core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java]: I pushed the new version and I believe it simplified things significantly. also it means LOCAL COPY work as a regular COPY query. Beside the JDBC test I also tested it with a Python Postgres client and it works nicely. 
```python
import psycopg2 as pg
conn = pg.connect(user="admin",
                                  password="quest",
                                  host="localhost",
                                  port="8812",
                                  database="qdb")
cursor = conn.cursor()

cursor.execute('drop table if exists try')
cursor.execute('COPY try FROM "core/src/test/resources/csv/test-numeric-headers.csv"')

cursor.close()
conn.close()
```

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java [hunk 3]
```diff
@@ -1679,6 +1681,32 @@ public void testCopyIn() throws SQLException {
         }
     }
```
Comment (bziobrowski) [inline core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java:1683]: Formatting is a bit weird here .
Do you use default intellij formatter ?

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java [hunk 4]
```diff
@@ -2213,32 +2212,6 @@ private void sendAndReset() throws PeerDisconnectedException, PeerIsSlowToReadEx
         responseAsciiSink.reset();
     }
```
Comment (bziobrowski) [inline core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:2214]: This method looks like it's meant for copy in - that is copy from client to server . Is it completely wrong ? 
I know we don't rally implement it but maybe we could just remove the call but keep the method for later ?

File: core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java (hunk 3 above)
Comment (jerrinot) [reply core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java:1683]: ha! well spotted! I didnt use IDE formatted, I tried to format it manually to keep it consistent with the rest of the code. it seems I failed :)

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java (hunk 4 above)
Comment (jerrinot) [reply core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:2214]: no, the method is not completely wrong. it's for the copy-in sub protocol. we don't support that. 

we support only a local copy, from a protocol perspective then looks like a regular query. 

I can keep it (unused) in the code and add some comments. I had been considering it, but then decided an unused private method was too weird and it's in git anyway. but it's true it's hard to discover it when it's just in git.

**Review by bziobrowski — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request merged**